### ISSUE TITLE
Matrix test multiple versions of multiple packages

### ISSUE THREAD

**Issue by torfsen** [watson/test-all-versions#3]
As far as I understand, it's currently only possible to test multiple versions a single package using a given command. Say run `npm test1` for 3 different versions of `dependency1`. It's also possible to add more than one such configuration, e.g. also run `npm test2` for 2 different versions of `dependency2`.

What's currently not possible (please correct me if I'm wrong) is to "matrix test" these 3 * 2 dependencies (using a single command): run `npm test_matrix` for

* `dependency1` (version 1) and `dependency2` (version 1)
* `dependency1` (version 1) and `dependency2` (version 2)
* `dependency1` (version 2) and `dependency2` (version 1)
* `dependency1` (version 2) and `dependency2` (version 2)
* `dependency1` (version 3) and `dependency2` (version 1)
* `dependency1` (version 3) and `dependency2` (version 2)

Basically, run my test suite for all possible combinations of supported dependency versions across multiple dependencies.

**Comment by watson** [comment]
Correct, this isn't possible at the moment.

I don't have time to add support for it right now unfortunately.
